spring-xd-module/src/main/java/org/springframework/xd/module/support/ModuleUtils.java

```diff
@@ -75,24 +73,29 @@ public static Resource modulePropertiesFile(SimpleModuleDefinition definition, C
 	}
 
 	/**
-	 * Return an expected module resource searching conventional file names. Will throw an exception if more than one
-	 * such resource exists
+	 * Return an expected module resource given a file extension. Will throw an exception if more than one such
+	 * resource
+	 * exists
 	 */
 	public static Resource locateModuleResource(SimpleModuleDefinition definition, ClassLoader moduleClassLoader,
 			String extension) {
-		String[] supportedFileNames = new String[] {definition.getName(), SPRING_MODULE};
-
+		PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver(moduleClassLoader);
 		Resource result = null;
 		String ext = extension.startsWith(".") ? extension : "." + extension;
-		for (String fileName : supportedFileNames) {
-			Resource candidate = new ClassPathResource(String.format("/%s/%s%s", "config", fileName, ext),
-					moduleClassLoader);
-			if (candidate.exists() && candidate.isReadable()) {
-				Assert.isNull(result, String.format("duplicate module definitions found: %s and %s",
-						(result == null ? "" : result.getFilename()), candidate.getFilename()));
-				result = candidate;
+		try {
+			Resource[] resources = resolver.getResources("/config/*" + ext);
```

Comment:
We need to be wary that using the `moduleClassLoader` as it is currently constructed, may return any classpath resource that is inside a `/config/` dir (_e.g._ if log4j.jar has `/config/foo.properties`, this **will** be returned).

Maybe we need to pass only a "top level module classloader" for that purpose